### PR TITLE
timer: timer inner loop auto recover from panic

### LINKED ISSUE #46411: timer framework should auto recover when some background loop panic

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #46411

Problem Summary:

There are some background goroutines running in timer framework.  To make the code robust, we should add some recover code in case it panics for some unknown reason.

### What is changed and how it works?

1. When the `loop` function in `TimerGroupRuntime` panics, it will sleep serval serval resume this function
2. When the `loop` function in `hookWorker` panics, it will sleep serval serval resume this function (hook will be recreated and call `Start` agagin)
3. When hook code panics, it will response to runtime with a fail result immediately. (hook will not be recreated)

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: timer/runtime/runtime.go [hunk 1]
```diff
@@ -145,12 +150,16 @@ func (rt *TimerGroupRuntime) Stop() {
...
-func (rt *TimerGroupRuntime) loop() {
-	rt.logger.Info("TimerGroupRuntime loop started")
-	defer func() {
-		rt.logger.Info("TimerGroupRuntime loop exit")
-		rt.wg.Done()
-	}()
+func (rt *TimerGroupRuntime) loop(totalPanic uint64) {
+	if totalPanic > 0 {
+		sleep(rt.ctx, rt.retryLoopWait)
+		rt.logger.Info("TimerGroupRuntime loop resumed from panic",
```
Comment (bb7133) [inline timer/runtime/runtime.go:156]: I think using `Warn` is better? Ditto for the rest of `panic` logs.

Comment (lcwangchao) [reply timer/runtime/runtime.go:156]: The panic have already been logged in `WithRecover` function, here is just to information that the loop resumed again

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**